phoenix-ios/phoenix-ios/views/send/SendView.swift

```diff
@@ -11,338 +8,858 @@ fileprivate var log = LoggerFactory.shared.logger(filename, .trace)
 fileprivate var log = LoggerFactory.shared.logger(filename, .warning)
 #endif
 
-struct SendView: MVIView {
+struct SendView: View {
+	
+	enum NavLinkTag: Hashable, CustomStringConvertible {
+		case LoginView(flow: SendManager.ParseResult_Lnurl_Auth)
+		case ValidateView(flow: SendManager.ParseResult)
+		
+		var description: String {
+			switch self {
+			case .LoginView(_)    : return "LoginView"
+			case .ValidateView(_) : return "ValidateView"
+			}
+		}
+	}
 	
 	enum Location {
 		case MainView
 		case ReceiveView
 	}
-	
 	let location: Location
 	
-	@StateObject var mvi: MVIState<Scan.Model, Scan.Intent>
+	@State var inputFieldText: String = ""
 	
-	@Environment(\.controllerFactory) var factoryEnv
-	var factory: ControllerFactory { return factoryEnv }
+	@State var imagePickerResult: PickerResult? = nil
+	
+	@State var autocompleteSuggestions: [String] = []
+	
+	@State var sortedContacts: [ContactInfo] = []
```

Comment:
I have a few questions about the list of contacts displayed (especially when the SendView is initialised)
- is this `sortedContacts` ordered by latest paid ones?
  - if it is, is the code shared or specific to the iOS app?
- do you display all the contacts, or just a subset (e.g., the last 5 contacts)?
  - if only a subset, how can the user access the list of all contacts ?

Reply:
The `sortedContacts` is the list of all contacts, sorted by name, as delivered by the database query.

The final plan is to have 2 different sections in the list:
- Recent Contacts
- All Contacts

Where "recent contacts" is maybe the 5 most recent that we've sent payments to.

Note that displaying "all contacts" here doesn't require any additional work. The list of contacts is already available in memory, and the UI `List` is "lazy", meaning it only bothers rendering what is visible on the screen.

Also, we can follow Apple's lead: in the Contacts app they display the alphabet on the trailing edge of the screen, making it super easy to jump to a letter, and find a certain person without typing.

However, we need to be smart about these features. For example, if the user only has a total of 5 contacts, it doesn't make sense to display the "recent contacts" section - only the "all contacts" section. And Apple doesn't bother to display the alphabet until the user has more than X total contacts. We should probably do something similar.

Reply:
Having 2 sections makes sense. For now, a list of all contacts is fine, especially if the list can be filtered in the text input.